### ISSUE TITLE
Profile switching is slow as NVDA freezes when connecting to a Braille device using Bluetooth

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#14399]
### Steps to reproduce:
1. Use Bluetooth to connect to a braille device, such as Seika NoteTaker;
2. Create new Profile for different applications;
3. Set one of the profile to "no-braille device";

### Actual behavior:
During switching between two profiles, NVDA freezes due to conecting for Bluetooth device;
### Expected behavior:
Bluetooth devices may take some time to connect, but NVDA should not freeze.
### NVDA logs, crash dumps and other attachments:
None
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2022.4Beta3 and Alpha
#### Windows version:
Windows 10 22H2 (x64) build 19045.2251
#### Name and version of other software in use when reproducing the issue:
None
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
None
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by cary-rowen** [comment]
@feerrenrut

**Comment by seanbudd** [comment]
It may be possible to retain the connection in the background, allowing profile switching / disabling a braille device., without formally ending the connection.
NVDA can just stop communicating braille information to the device when braille is disabled.
This concept would be similar to keeping a bluetooth headphone connection, but muting it.

**Comment by cary-rowen** [comment]
@seanbudd 
Will this be easy to implement?

@moyanming do you have any ideas?

**Comment by feerrenrut** [comment]
>It may be possible to retain the connection in the background, allowing profile switching / disabling a braille device., without formally ending the connection.

One obstacle will be differentiating a temporary change, E.G., a profile switch, from a standard config change, E.G., setting 'no braille' in the user settings GUI.


Another way to improve this would be to look into making this non-blocking for NVDA while waiting on the connection to be established.

First, this particular case needs to be investigated to understand what causes it, to ensure there isn't something specific to this braille driver that could be done differently.

**Comment by Adriani90** [comment]
@cary-rowen how is it now in the last NVDA alpha version? Are you still having this issue?

cc: @burmancomp

**Comment by SaschaCowley** [comment]
Note: This does not seem to be reproducible with automatic display detection; the display has to be chosen explicitly.

**Comment by Adriani90** [comment]
@SaschaCowley are you saying you are able to reproduce when you choose a braille display explicitely? Is this still the case in last NVDA alpha?
Is this specific to a certain braille display?
cc: @LeonarddeR

**Comment by SaschaCowley** [comment]
@Adriani90 
> are you saying you are able to reproduce when you choose a braille display explicitely?

Yes, exactly. If I choose my display explicitly, the freeze occurs; if I select "Automatic", it does not (though there is a delay in connecting to the display).

This is because automatic display detection takes place in a background thread.

> Is this still the case in last NVDA alpha?

I just tested it. Unfortunately I'm currently running a custom build for testing, but I'm only 4 merges behind master, and I doubt any of the commits would make a difference.

```psh
PS D:\projects\nvda> git log --oneline master ^$(git merge-base master fixSapi4)                                        
a9a081b6b (HEAD -> master, origin/master) Remove 'Automatically set system focus to focusable elements' (#17598)        
098863bd3 Enable whole-program optimization when compiling eSpeak-NG (#17631)                                           
fa619c2d5 MS Word documents: report when headings are collapsed in both speech and braille. (#17499)                    
0be66d5fd Split Java Access Bridge calls to fix NVDA crashes when selecting whole text in specific source files in Android studio (#17604)                                                                                                      
```

> Is this specific to a certain braille display?

I doubt it, but I'm using a Hims Brailleedge 40 (software version 1.1 build 20170307).

**Comment by LeonarddeR** [comment]
When profile switching and initializing a driver explicitly, when it is threadsafe, we should initialize the driver on a background thread. That shouldn't be a very big deal. I believe that since this issue was initially opened, code has already been written to minimize the cases where profile switching would result in reconnecting when the drivers of the old and new profile are equal.
